### PR TITLE
Refactor the `contrib` submodule and add equivalent `NormalNoise` base models

### PR DESCRIPTION
**Pull request by michaelosthege**
### Motivation
We have three really useful base models with `StudentTNoise`, but sometimes a `NormalNoise` model is already sufficient.

This PR refactors the `calibr8.contrib` submodule and adds base models such that we have the same base models available for `NormalNoise` that we have for `StudenttNoise`.

### What changed
* `calibr8.contrib.base` became `calibr8.contrib.studentt`
* `calibr8.contrib.normal` was introduced

### CODE REVIEW

File: calibr8/__init__.py [hunk 1]
```diff
@@ -6,7 +6,12 @@
     StudentTNoise,
     PoissonNoise,
 )
-from .contrib.base import (
+from .contrib.normal import (
+    BaseAsymmetricLogisticN,
+    BaseLogIndependentAsymmetricLogisticN,
+    BasePolynomialModelN,
```
Comment (lhelleckes) [inline calibr8/__init__.py:12]: If we want to do it the same way `StudentT` was implemented, a `BaseModelN `is missing here.

File: calibr8/contrib/normal.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+
+class BasePolynomialModelN(core.ContinuousUnivariateModel, noise.NormalNoise):
+    def __init__(
+        self, *,
+        independent_key: str, dependent_key: str,
+        mu_degree: int, sigma_degree: int=0,
+        theta_names: Optional[Tuple[str]]=None,
+    ):
+        """ Template for a model with polynomial trend (mu) and sigma (as a function of mu).
```
Comment (lhelleckes) [inline calibr8/contrib/normal.py:18]: This descirption could be valid for the Student-t version as well. Should we not mention the Normal distribution in the description?

**Review by lhelleckes — CHANGES_REQUESTED** [review]
I see some inconsistencies compared to the `studentt` module. The `BaseModelT` should be removed or a `BaseModelN` added to the new module i.m.o.

File: calibr8/__init__.py (hunk 1 above)
Comment (michaelosthege) [reply calibr8/__init__.py:12]: The `BaseModelT` is an artifact of the old class/inheritance design.
I'd like to get rid of it wihout breaking the API though. I'll check if we can add a `DeprecationWarning` for any code that uses it.

**Comment by michaelosthege** [comment]
@lhelleckes I added a `FutureWarning` for the `BaseModelT`, because the `DeprecationWarning` is not shown by default.
Please review these additions :)

**Review by lhelleckes — APPROVED** [review]
Good changes, thanks!

**Pull request merged**